ddtrace/contrib/pytest/plugin.py

```diff
@@ -118,7 +118,10 @@ def pytest_runtest_protocol(item, nextitem):
         span.set_tag(SPAN_KIND, KIND)
         span.set_tag(test.FRAMEWORK, FRAMEWORK)
         span.set_tag(test.NAME, item.name)
-        span.set_tag(test.SUITE, item.module.__name__)
+        if hasattr(item, "module"):
+            span.set_tag(test.SUITE, item.module.__name__)
+        elif hasattr(item, "dtest"):
+            span.set_tag(test.SUITE, item.dtest.globs["__name__"])
```

Comment:
I was wondering if we should have guards here around but it would appear that `globs` in the case of a doctest is a copy of the module's `__dict__`.

https://github.com/python/cpython/blob/beb3a835dae3cd940b93e7caa32450890c4cd539/Lib/doctest.py#L924-L935

In the spirit of ducktyping, we could play it safe and add the other property and dict checks before setting a tag.

@Yun-Kim What do you think?

Reply:
I agree about playing it safe, but what do you mean by other property? 
```python
elif hasattr(item, "dtest") and item.dtest.globs.get("__name__")
```
Do you mean something like this?

Reply:
The complete chain would be:

```python
elif hasattr(item, "dtest") and hasattr(item.dtest, "globs") and item.dtest.globs.get("__name__"):
```

Reply:
what about... `hasattr(item, "dtest") and isinstance(item.dtest, DocTest)` ?

(is `item` the `DocTest`?)

Reply:
The `item` in this case is a `DoctestItem`, which contains a `Doctest` object stored as `item.dtest`. It's this `dtest` object which contains the `glob` dictionary with the module name.

Reply:
@majorgreys from your provided link it looks like the globs object will always have a `__name__` key value pair, it seems like we can skip that?

Reply:
> The `item` in this case is a `DoctestItem`, which contains a `Doctest` object stored as `item.dtest`. It's this `dtest` object which contains the `glob` dictionary with the module name.

Can we do a `isinstance(item, DoctestItem)` and that is enough?

Reply:
That would be my first choice, but it looks like `DoctestItem` isn't part of pytest's public API and I can't import the type `DoctestItem` 😢

Reply:
ah, shoot, you can import the `Doctest` class, so could do `if isinstance(getattr(item, "dtest", None), Doctest):` if the API of `Doctest.globs["__name__"]` is stable enough